cli/lsp/documents.rs

```diff
@@ -252,6 +262,7 @@ fn get_maybe_test_module_fut(
 pub struct DocumentOpenData {
   lsp_version: i32,
   maybe_parsed_source: Option<ParsedSourceResult>,
+  maybe_semantic_tokens: Arc<Mutex<Option<lsp::SemanticTokens>>>,
```

Comment:
Might be nice to add something like `CloneCell<T: Send + Clone>` to `cli::util::sync` that has methods like `set(value: T)` and `get_cloned(&self): T` in order to bury the mutex there.

Reply:
That said, should this use something like OnceCell instead? I guess no because async.